### DISCUSSION TITLE
Questions about blockchain

### DISCUSSION THREAD

**Discussion by zyf654321** [HewlettPackard/swarm-learning#78] [Q&A]
Hello, I read your paper. But there is very little talk in the paper about the blockchain design of swarm learning.  how can I learn in detail about the design of swarm learning's blockchain? Sorry to bother you.

**Comment by RadhakrishnaJ** [comment]
The Swarm Network nodes interact with each other using this blockchain platform to maintain global state information about the model that is being trained and to track progress (note that only metadata is written to the blockchain. The model itself is not stored in the blockchain.) The Swarm Network nodes use this state and progress information to coordinate the working of the Swarm Learning nodes.

Blockchain maintains the “global state” of Swarm learning training. It contains set of Smart contracts (Smart contract = data member+ methods to operate on data members). In simple words it contains information of decentralized State of Swarm Learning training progress. 

Refer [Swarm components interaction](https://github.com/HewlettPackard/swarm-learning/blob/master/docs/Component_interactions.md#component-interactions) for interaction details. 
 
ML model parameters (weights & biases)  not shared with blockchain. SL nodes uses a file server to share ML model parameters among themselves directly.

**Reply by PNg-HA** [reply]
Hi. How could I read the blockchain smart contract record (or results) after training completion.
